UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -246,6 +236,20 @@ public AgentInfo Info
 
         WriteAdapter m_WriteAdapter = new WriteAdapter();
 
+
+        /// Represents the reward the agent accumulated during the current step.
+        /// It is reset to 0 at the beginning of every step.
+        /// Should be set to a positive value when the agent performs a "good"
+        /// action that we wish to reinforce/reward, and set to a negative value
+        /// when the agent performs a "bad" action that we wish to punish/deter.
+        /// Additionally, the magnitude of the reward should not exceed 1.0
+        LowLevelRewardProviderComponent m_RewardProviderComponent;
```

Comment:
Why does this reward provider need to be a component ? I suppose this means there will not always be this LowLevelRewardProvider attached to the Agent and that it will be optional. I wonder if we should alway give this reward provider as a base with possibility to add others or if this base one should be manually added by the user...

Reply:
This is just the reward provider component wrapper.  If users create their own and add them to the Agent, that is something we want to support.